near-sdk/tests/test-contracts/store/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+
+        Self {
+            vec,
+            iterable_set,
+            iterable_map,
+            unordered_set,
+            unordered_map,
+            tree_map,
+            lookup_map,
+            lookup_set,
+        }
+    }
+
+    #[payable]
+    pub fn exec(&mut self, col: Collection, op: Op) {
+        match op {
+            Op::Insert(val) => self.insert(col, val),
+            Op::Remove(val) => self.remove(col, val),
+            Op::Flush => self.flush(col),
+            Op::Contains(val) => self.contains(col, val),
+            Op::Iter(take) => self.iter(col, take),
+        }
+    }
+
+    fn insert(&mut self, col: Collection, val: u32) {
+        match col {
+            IterableSet => {
+                self.iterable_set.insert(val);
+            }
+            IterableMap => {
+                self.iterable_map.insert(val, val);
+            }
+            UnorderedMap => {
+                self.unordered_map.insert(val, val);
+            }
+            UnorderedSet => {
+                self.unordered_set.insert(val);
+            }
+            LookupMap => {
+                self.lookup_map.insert(val, val);
+            }
+            LookupSet => {
+                self.lookup_set.insert(val);
+            }
+            TreeMap => {
+                self.tree_map.insert(val, val);
+            }
+            Vector => {
+                self.vec.push(val);
+            }
+        };
+    }
+
+    fn remove(&mut self, col: Collection, val: u32) {
+        match col {
+            IterableSet => {
+                self.iterable_set.remove(&val);
+            }
+            IterableMap => {
+                self.iterable_map.remove(&val);
+            }
+            UnorderedMap => {
+                self.unordered_map.remove(&val);
+            }
+            UnorderedSet => {
+                self.unordered_set.remove(&val);
+            }
+            LookupMap => {
+                self.lookup_map.remove(&val);
+            }
+            LookupSet => {
+                self.lookup_set.remove(&val);
+            }
+            TreeMap => {
+                self.tree_map.remove(&val);
+            }
+            Vector => {
+                if self.vec.is_empty() {
+                    return;
+                }
+                // Take the opportunity to take swap and pop.
+                self.vec.swap_remove(self.vec.len() - 1);
+            }
+        };
+    }
+
+    fn flush(&mut self, col: Collection) {
+        match col {
+            IterableSet => self.iterable_set.flush(),
+            IterableMap => self.iterable_map.flush(),
+            UnorderedMap => self.unordered_map.flush(),
+            UnorderedSet => self.unordered_set.flush(),
+            LookupMap => self.lookup_map.flush(),
+            TreeMap => self.tree_map.flush(),
+            Vector => self.vec.flush(),
+            // No flush method.
+            LookupSet => unimplemented!(),
+        };
+    }
```

Comment:
This is pointless as a benchmark test. If there are no changes, there is nothing to flush.

Reply:
This is executed after insert tests.

Reply:
Flush tests don't work even with pre-populated data, because `env::storage_write` does not charge any gas. I'd propose to ditch that for now.